apm-sniffer/apm-sdk-plugin/spring-plugins/core-patch/src/main/java/org/apache/skywalking/apm/plugin/spring/patch/CreateAopProxyInterceptor.java

```diff
@@ -32,25 +34,45 @@ public class CreateAopProxyInterceptor implements InstanceMethodsAroundIntercept
 
     @Override
     public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
-        MethodInterceptResult result) throws Throwable {
+                             MethodInterceptResult result) throws Throwable {
 
     }
 
     @Override
     public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
-        Object ret) throws Throwable {
+                              Object ret) throws Throwable {
         AdvisedSupport advisedSupport = (AdvisedSupport) allArguments[0];
 
-        Class targetClass = advisedSupport.getTargetClass();
-        if (targetClass != null && EnhancedInstance.class.isAssignableFrom(targetClass)) {
-            return true;
+        if (maybeHasUserSuppliedProxyInterfaces(ret)) {
+            Class targetClass = advisedSupport.getTargetClass();
+            if (targetClass != null) {
+                if (onlyImplementsEnhancedInstance(advisedSupport) || onlyImplementsEnhancedInstanceAndSpringProxy(advisedSupport)) {
+                    return true;
+                }
+            }
         }
         return ret;
     }
 
+    private boolean maybeHasUserSuppliedProxyInterfaces(Object ret) {
+        return !(Boolean) ret;
+    }
+
+    private boolean onlyImplementsEnhancedInstanceAndSpringProxy(AdvisedSupport advisedSupport) {
+        Class<?>[] ifcs = advisedSupport.getProxiedInterfaces();
+        Class targetClass = advisedSupport.getTargetClass();
+        return ifcs.length == 2 && EnhancedInstance.class.isAssignableFrom(targetClass) && SpringProxy.class.isAssignableFrom(targetClass);
```

Comment:
When is `SpringProxy.class` added into the hierarchy tree?

Reply:
Thank you for the review.
As shown below, the `hasNoUserSuppliedProxyInterfaces` method in `DefaultAopProxyFactory` is defined as follows:
**spring4/spring5**
```
         /**
	 * Determine whether the supplied {@link AdvisedSupport} has only the
	 * {@link org.springframework.aop.SpringProxy} interface specified
	 * (or no proxy interfaces specified at all).
	 */
	private boolean hasNoUserSuppliedProxyInterfaces(AdvisedSupport config) {
		Class<?>[] ifcs = config.getProxiedInterfaces();
		return (ifcs.length == 0 || (ifcs.length == 1 && SpringProxy.class.isAssignableFrom(ifcs[0])));
	}
```
**spring3**
```
         /**
	 * Determine whether the supplied {@link AdvisedSupport} has only the
	 * {@link org.springframework.aop.SpringProxy} interface specified
	 * (or no proxy interfaces specified at all).
	 */
	private boolean hasNoUserSuppliedProxyInterfaces(AdvisedSupport config) {
		Class[] interfaces = config.getProxiedInterfaces();
		return (interfaces.length == 0 || (interfaces.length == 1 && SpringProxy.class.equals(interfaces[0])));
	}
```

Reply:
OK, this seems ok. Could you update the changelog? Then I could merge this PR.

Reply:
Thank you for the review. 
The `Changes.md` file has been updated.